<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>ThreadsSync</class>
 <widget class="QMainWindow" name="ThreadsSync">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>760</width>
    <height>600</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>ThreadsSync</string>
  </property>
  <widget class="QWidget" name="centralwidget">
   <widget class="QPushButton" name="create">
    <property name="geometry">
     <rect>
      <x>40</x>
      <y>490</y>
      <width>151</width>
      <height>41</height>
     </rect>
    </property>
    <property name="text">
     <string>Create threads</string>
    </property>
   </widget>
   <widget class="QComboBox" name="chosenTask">
    <property name="geometry">
     <rect>
      <x>210</x>
      <y>490</y>
      <width>151</width>
      <height>41</height>
     </rect>
    </property>
    <item>
     <property name="text">
      <string>Default</string>
     </property>
    </item>
    <item>
     <property name="text">
      <string>Find minimum</string>
     </property>
    </item>
   </widget>
   <widget class="QGroupBox" name="groupBox">
    <property name="geometry">
     <rect>
      <x>530</x>
      <y>20</y>
      <width>191</width>
      <height>281</height>
     </rect>
    </property>
    <property name="title">
     <string>Controls</string>
    </property>
    <widget class="QPushButton" name="Pause">
     <property name="geometry">
      <rect>
       <x>20</x>
       <y>180</y>
       <width>151</width>
       <height>41</height>
      </rect>
     </property>
     <property name="text">
      <string>Pause</string>
     </property>
    </widget>
    <widget class="QPushButton" name="run">
     <property name="geometry">
      <rect>
       <x>20</x>
       <y>130</y>
       <width>151</width>
       <height>41</height>
      </rect>
     </property>
     <property name="text">
      <string>Run</string>
     </property>
    </widget>
    <widget class="QComboBox" name="priority">
     <property name="geometry">
      <rect>
       <x>20</x>
       <y>40</y>
       <width>151</width>
       <height>41</height>
      </rect>
     </property>
     <item>
      <property name="text">
       <string>Idle</string>
      </property>
     </item>
     <item>
      <property name="text">
       <string>Below normal</string>
      </property>
     </item>
     <item>
      <property name="text">
       <string>Normal</string>
      </property>
     </item>
     <item>
      <property name="text">
       <string>Above normal</string>
      </property>
     </item>
     <item>
      <property name="text">
       <string>High</string>
      </property>
     </item>
    </widget>
    <widget class="QPushButton" name="killThread">
     <property name="geometry">
      <rect>
       <x>20</x>
       <y>230</y>
       <width>151</width>
       <height>41</height>
      </rect>
     </property>
     <property name="text">
      <string>Kill</string>
     </property>
    </widget>
   </widget>
   <widget class="QTableWidget" name="availableThreads">
    <property name="geometry">
     <rect>
      <x>40</x>
      <y>20</y>
      <width>471</width>
      <height>381</height>
     </rect>
    </property>
   </widget>
   <widget class="QTextEdit" name="time">
    <property name="geometry">
     <rect>
      <x>550</x>
      <y>490</y>
      <width>151</width>
      <height>41</height>
     </rect>
    </property>
   </widget>
   <widget class="QPushButton" name="runall">
    <property name="geometry">
     <rect>
      <x>550</x>
      <y>440</y>
      <width>151</width>
      <height>41</height>
     </rect>
    </property>
    <property name="text">
     <string>Get running time</string>
    </property>
   </widget>
   <widget class="QLabel" name="label">
    <property name="geometry">
     <rect>
      <x>490</x>
      <y>490</y>
      <width>61</width>
      <height>41</height>
     </rect>
    </property>
    <property name="text">
     <string>Took:</string>
    </property>
   </widget>
   <widget class="QLabel" name="label_2">
    <property name="geometry">
     <rect>
      <x>40</x>
      <y>410</y>
      <width>131</width>
      <height>41</height>
     </rect>
    </property>
    <property name="text">
     <string>Number of iterations:</string>
    </property>
   </widget>
   <widget class="QTextEdit" name="steps">
    <property name="geometry">
     <rect>
      <x>40</x>
      <y>440</y>
      <width>151</width>
      <height>41</height>
     </rect>
    </property>
   </widget>
   <widget class="QComboBox" name="numOfThreads">
    <property name="geometry">
     <rect>
      <x>210</x>
      <y>440</y>
      <width>151</width>
      <height>41</height>
     </rect>
    </property>
    <item>
     <property name="text">
      <string>1</string>
     </property>
    </item>
    <item>
     <property name="text">
      <string>2</string>
     </property>
    </item>
    <item>
     <property name="text">
      <string>4</string>
     </property>
    </item>
    <item>
     <property name="text">
      <string>8</string>
     </property>
    </item>
    <item>
     <property name="text">
      <string>16</string>
     </property>
    </item>
   </widget>
   <widget class="QComboBox" name="syncType">
    <property name="geometry">
     <rect>
      <x>380</x>
      <y>440</y>
      <width>151</width>
      <height>41</height>
     </rect>
    </property>
    <item>
     <property name="text">
      <string>Interlock</string>
     </property>
    </item>
    <item>
     <property name="text">
      <string>Critical Section</string>
     </property>
    </item>
    <item>
     <property name="text">
      <string>Mutex</string>
     </property>
    </item>
    <item>
     <property name="text">
      <string>Semaphore</string>
     </property>
    </item>
   </widget>
  </widget>
  <widget class="QMenuBar" name="menubar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>760</width>
     <height>21</height>
    </rect>
   </property>
  </widget>
  <widget class="QStatusBar" name="statusbar"/>
 </widget>
 <resources/>
 <connections/>
</ui>
